autoflip_calculators/text_detection_calculator.cc

```diff
@@ -0,0 +1,262 @@
+  return pixel / static_cast<double>(frame_height_);
+}
+
+void TextDetectionCalculator::DecodeBoundingBoxes(const Mat& scores, const Mat& geometry, float scoreThresh,
+                         std::vector<RotatedRect>& detections, std::vector<float>& confidences) {
+    detections.clear();
+    RET_CHECK(scores.dims == 4)
+        << "Scores' dimension must be 4.";
+    RET_CHECK(scores.size[0] == 1)
+        << "scores.size[0] must be 1.";
+    RET_CHECK(scores.size[1] == 1)
+        << "scores.size[1] must 1.";
+    RET_CHECK(geometry.dims == 4)
+        << "Geometry dimension must be 4.";
+    RET_CHECK(geometry.size[0] == 1)
+        << "geometry.size[0] must be 1.";
+    RET_CHECK(geometry.size[1] == 5)
+        << "geometry.size[1] must be 5.";
+    RET_CHECK(scores.size[2] == geometry.size[2])
+        << "scores.size[2] must equal to geometry.size[2].";
+    RET_CHECK(scores.size[3] == geometry.size[3])
+        << "scores.size[3] must equal to geometry.size[3].";
+
+    const int height = scores.size[2];
+    const int width = scores.size[3];
+    for (int y = 0; y < height; ++y) {
+        const float* scoresData = scores.ptr<float>(0, 0, y);
+        const float* x0_data = geometry.ptr<float>(0, 0, y);
+        const float* x1_data = geometry.ptr<float>(0, 1, y);
+        const float* x2_data = geometry.ptr<float>(0, 2, y);
+        const float* x3_data = geometry.ptr<float>(0, 3, y);
+        const float* anglesData = geometry.ptr<float>(0, 4, y);
+
+        for (int x = 0; x < width; ++x) {
+            float score = scoresData[x];
+            if (score < scoreThresh)
+                continue;
+            
+            // Decode a prediction.
+            // Multiple by 4 because maps are 4 time less than input image.
+            float offsetX = x * 4.0f, offsetY = y * 4.0f;
+            float angle = anglesData[x];
+            float cosA = std::cos(angle);
+            float sinA = std::sin(angle);
+            float h = x0_data[x] + x2_data[x];
+            float w = x1_data[x] + x3_data[x];
+
+            cv::Point2f offset(offsetX + cosA * x1_data[x] + sinA * x2_data[x],
+                           offsetY - sinA * x1_data[x] + cosA * x2_data[x]);
+            cv::Point2f p1 = Point2f(-sinA * h, -cosA * h) + offset;
+            cv::Point2f p3 = Point2f(-cosA * w, sinA * w) + offset;
+
+            cv::RotatedRect r(0.5f * (p1 + p3), Size2f(w, h), -angle * 180.0f / (float)CV_PI);
+            detections.push_back(r);
+            confidences.push_back(score);
+        } // end for x
+    } // end for y
+}   
+
+::mediapipe::Status TextDetectionCalculator::Process(
+    mediapipe::CalculatorContext* cc) {
+  // What if !cc->Inputs.HasTag("VIDEO")
+  if (cc->Inputs().HasTag("VIDEO") &&
+      cc->Inputs().Tag("VIDEO").Value().IsEmpty()) {
+    return ::mediapipe::UnknownErrorBuilder(MEDIAPIPE_LOC)
+           << "No VIDEO input at time " << cc->InputTimestamp().Seconds();
+  }
+
+  if (cc->Inputs().HasTag("VIDEO")) {
```

Comment:
See above comment, probably don't make video input optional so this if statement is not needed.